enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,191 @@
+# Adding Exclusive Key in ManagedClusterSet
+ 
+## Release Signoff Checklist
+ 
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+ 
+## Summary
+The proposed work enhances the managedClusterSet API to support managedClusterSet Override.
+ 
+## Motivation
+Today, managedClusterSet is designed to be exclusive. A managedCluster can only be included in only one managedClusterSet, by setting a label `cluster.open-cluster-management.io/clusterset` on the managedCluster. A user must have explicit RBAC rules, `managedclustersets/join`, to add such labels.
+ 
+There are also raising requirements on using managedClusterSet for the following purposes. Within a purpose, the managedClusterSet must be exclusive, but for different purposes it can overlap.
+- I want to create a managedClusterSet which includes all clusters in one region. Then I can apply certain configurations to all clusters in the region.
+- I want to create a managedClusterSet for Disaster Recovery purposes, the managedClusterSet should select two or more clusters in different regions.
+- I want to create a managedClusterSet to group clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+ 
+## Proposal
+1. Changes on ManagedClusterSet APIs.
+ 
+Add a field into ManagedClusterSet spec
+ 
+```go
+type ManagedClusterSetSpec struct {
+ // ExclusiveKey defines the key of the label which the managedClusterSet will use to select managedCluster.
+ // The managedClusterSet will select ManagedClusters based on the managedCluster's label "<ExclusiveKey>:<ManagedClusterSetName>".
```

Comment:
Why is this more desireable than a general label selector?

Reply:
Will we run into an issue with identical values in label key:value pairs and ManagedClusterSet being cluster-scoped;

for example:

foo=bar
MCS name = bar, exclusiveKey = foo

other=bar
MCS name = bar, exclusiveKey = other

Reply:
We have changed the spec, and use `clusterLabel` to select managedClusters.

We need to make sure that, managedClusterSets should be exclusive within a purpose, but for different purposes it can overlap.